modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SlaCard/index.js

```diff
@@ -0,0 +1,200 @@
+	const [slaSelected, setSlaSelected] = useState('');
+
+	useEffect(() => {
+		const fetchkoroneikeAccount = async () => {
+			const {data} = await client.query({
+				query: getKoroneikiAccounts,
+				variables: {
+					filter: `accountKey eq '${accountKey}'`,
+				},
+			});
+			if (data) {
+				const dataPath = data.c.koroneikiAccounts.items[0];
+				const formatDate = {
+					day: '2-digit',
+					month: '2-digit',
+					year: 'numeric',
+				};
+				const slaFiltedData = [];
+				const slaName = {
+					current: 'Current',
+					expired: 'Expired',
+					future: 'Future',
+				};
+				const slaPathLabel = {
+					current: slaName.current.toUpperCase(),
+					expired: slaName.expired.toUpperCase(),
+					future: slaName.future.toUpperCase(),
+				};
+
+				const slaNewDate = (sla) => {
+					return new Date(dataPath[sla]).toLocaleDateString(
+						'en-US',
+						formatDate
+					);
+				};
+				const slaGetData = (sla) => {
+					const slaFullTitle = 'sla' + sla;
+
+					return {
+						slaDateEnd: `${slaNewDate(slaFullTitle + 'EndDate')}`,
+						slaDateStart: `${slaNewDate(
+							slaFullTitle + 'StartDate'
+						)}`,
+						slaLabel: sla.toUpperCase(),
+						slaTitle: dataPath[slaFullTitle].split(' ')[0],
+					};
+				};
+
+				const slaRawData = [
+					slaGetData(slaName.current),
+					slaGetData(slaName.expired),
+					slaGetData(slaName.future),
+				].filter((sla) => sla.slaTitle);
+				const slaRaw = {
+					first: slaRawData[0],
+					second: slaRawData[1],
+					third: slaRawData[2],
+				};
+				if (
+					slaRaw.first?.slaLabel === slaPathLabel.current &&
+					(slaRaw.second?.slaTitle || slaRaw.third?.slaTitle) ===
+						slaRaw.first?.slaTitle
+				) {
+					slaFiltedData.push(slaRaw.first);
+
+					if (
+						slaFiltedData[0].slaTitle === slaRaw.second.slaTitle &&
+						slaRaw.second.slaLabel === slaPathLabel.expired
+					) {
+						slaFiltedData[0].slaDateStart =
+							slaRaw.second.slaDateStart;
+					}
+					if (
+						slaFiltedData[0].slaTitle === slaRaw.second.slaTitle &&
+						slaRaw.second.slaLabel === slaPathLabel.future
+					) {
+						slaFiltedData[0].slaDateEnd = slaRaw.second.slaDateEnd;
+					}
+					if (slaRaw.third) {
+						if (
+							slaRaw.third.slaTitle === slaFiltedData[0].slaTitle
+						) {
+							slaFiltedData[0].slaDateEnd =
+								slaRaw.third.slaDateEnd;
+						} else {
+							slaFiltedData.push(slaRaw.third);
+						}
+					}
+				} else {
+					slaRawData.map((sla) => slaFiltedData.push(sla));
+				}
+				setSlaData(slaFiltedData);
+
+				if (!slaSelected) {
+					const slaSelectedCard = slaFiltedData[0].slaLabel;
+					setSlaSelected(slaSelectedCard);
+				}
+			}
+		};
+
+		fetchkoroneikeAccount();
+	}, [accountKey, slaSelected]);
+
+	const handleClick = () => {
+		const slaLength = slaData.length;
+		if (slaSelected === slaData[slaLength - 1].slaLabel) {
+			setSlaSelected(slaData[0].slaLabel);
+		} else if (slaSelected === slaData[0].slaLabel) {
```

Comment:
Why did you not verify if [ slaData[ ] ] exists like you did in [slow.first?.sylLable]? Can you explain me? I didn't understand.

Reply:
I forgot @joanagrego , but this don't break the code, because if don't have any data on slaData, what I save on slaSelected don't will be use, whatever if this get empty string.